Added tests for node id reuse

lutovich opened the pull request:
To make sure it is possible to specify `NODE` as well as `RELATIONSHIP` in
`dbms.ids.reuse.types.override` setting.

changelog: [enterprise] Made possible to allow node id reuse using 'dbms.ids.reuse.types.override' setting.

lutovich committed:
Added tests for node id reuse

To make sure it is possible to specify `NODE` as well as `RELATIONSHIP` in
`dbms.ids.reuse.types.override` setting.

tinwelint commented on enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/configuration/EnterpriseEditionSettings.java:
```diff
@@ -32,7 +32,7 @@
 public class EnterpriseEditionSettings
 {
     @Description( "Specified names of id types (comma separated) that should be reused. " +
-                  "Currently only 'RELATIONSHIP' type is supported. " )
```
Where is the actual list of supported types enforced?

lutovich replied:
It is not enforced. I'll add this.

tinwelint replied:
Will this PR then enable `NODE` to be used? I'd expect not since it's merely adding tests.

lutovich replied:
Right, it will not enable node id reuse. It might not be entirely safe to enable node/relationship id reuse in 2.3. We should probably do it for 3.1.

lutovich replied:
@tinwelint added restriction on values in the list

tinwelint replied:
Oh, so there was previously nothing preventing config from specifying NODE or any other id type?

lutovich replied:
Yes right, sorry, missed your message somehow.

lutovich committed:
Restrict idTypesToReuse setting to 'node' and 'relationship'

Only node and relationship id reuse is supported. This commit makes the
corresponding setting accept list of only those tow values.
